TypeSystem/flow-analysis/reachability_break_A05_t01.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Break statement: If `N` is a statement of the form `break [L];`,
+/// then:
+/// - Let `S` be the statement targeted by the `break`. If `L` is not present,
+///   this is the innermost `do`, `for`, `switch`, or `while` statement.
+///   Otherwise it is the `do`, `for`, `switch`, or `while` statement with a
+///   label matching L.
+/// - Update `break(S) = join(break(S), before(N))`.
+/// - Let `after(N) = unreachable(before(N))`.
+///
+/// @description Checks that is some type `T` is made a type of interest
+/// `before(N)` then the variable can be promoted to `T` in `after(N)`.
+/// @author sgrekhov22@gmail.com
+
+class S {}
+
+class T extends S {
+  int answer() => 42;
+}
+
+test1() {
+  S s = S();
+  for (int j = 0; j < 1; j++) {
+    if (s is T) {} // Make `T` a type of interest
+    break;
+    s = T();
+    s.answer();
+  }
+}
+
+test2() {
+  S s = S();
+  for (var j in []) {
+    if (s is T) {}
+    break;
+    s = T();
```

Comment:
As usual, I think it would be nice to keep track of test cases that are concerned with the properties of dead code. Perhaps:

```suggestion
    // Unreachable, but does support promotion.
    s = T();
```

It should be fine if there's one such comment in each of the tests that are dealing with unreachable code.

Reply:
Thanks!